test/functional/uri_tests.js

```diff
@@ -164,4 +159,22 @@ describe('URI', function() {
       );
     }
   });
+
+  it('should correctly translate uri options using new parser', {
+    metadata: { requires: { topology: 'replicaset' } },
+    test: function(done) {
+      const config = this.configuration;
+      const uri = `mongodb://${config.host}:${config.port}/${config.db}?replicaSet=${
```

Comment:
Does this url specification work for a replicaset? I'd think we'd want to specify all three host:port combos.

Reply:
Yes it does. In fact, because of SDAM you only ever need to provide a single seed in the URI, and we can depend on discovery to find the rest. Only when you're connecting to mongos' do you need to explicitly list all of them out.